SDL build fails with ENABLE_LOG=OFF

### Bug Report
SDL build fails with ENABLE_LOG=OFFF

##### Reproduction Steps
1. Try to build SDL with cmake ` -ENABLE_LOG=OFF` flag

##### Expected Behavior
SDL is built successfully

##### Observed Behavior
Build failed

##### OS & Version Information
* OS/Version: Ubuntu 18.04
* SDL Core Version: https://github.com/smartdevicelink/sdl_core/tree/develop ( 1cd41269)

##### Test Case, Sample Code, and / or Example App
- Console log
```
12:38:15 [  5%] Building CXX object src/components/utils/CMakeFiles/Utils.dir/src/convert_utils.cc.o
12:38:16 /home/jenkins/workspace/Develop_SDL_NoUT_P_ENABLE_LOG_OFF/src/components/utils/src/convert_utils.cc: In function ‘std::__cxx11::string utils::ConvertBinaryDataToString(const uint8_t*, size_t)’:
12:38:16 /home/jenkins/workspace/Develop_SDL_NoUT_P_ENABLE_LOG_OFF/src/components/utils/src/convert_utils.cc:81:15: error: aggregate ‘std::locale loc’ has incomplete type and cannot be defined
12:38:16    std::locale loc;
12:38:16                ^~~
12:38:16 src/components/utils/CMakeFiles/Utils.dir/build.make:158: recipe for target 'src/components/utils/CMakeFiles/Utils.dir/src/convert_utils.cc.o' failed
12:38:16 make[2]: *** [src/components/utils/CMakeFiles/Utils.dir/src/convert_utils.cc.o] Error 1
```

--- a/src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/rc_app_extension.cc
+++ b/src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/rc_app_extension.cc
@@ -198,15 +198,13 @@ void RCAppExtension::RevertResumption(
   const auto module_subscriptions =
       ConvertSmartObjectToModuleCollection(resumption_data);
 
-  for (auto& module : module_subscriptions) {
-    SDL_LOG_TRACE("Requested to unsubscribe module_type  "
-                  << module.first << "module_id: " << module.second);
-  }
   std::set<rc_rpc_plugin::ModuleUid> to_be_unsubscribed;
 
   const auto app_id = application_.app_id();
   auto no_apps_subscribed = [app_id,
                              this](const rc_rpc_plugin::ModuleUid& module) {
+    SDL_LOG_TRACE("Requested to unsubscribe module_type  "
+                  << module.first << "module_id: " << module.second);
     if (plugin_.IsOtherAppsSubscribed(module, app_id)) {
       SDL_LOG_DEBUG("Some other app except " << app_id
                                              << " is already subscribed to "

--- a/src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/waypoints_pending_resumption_handler.cc
+++ b/src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/waypoints_pending_resumption_handler.cc
@@ -154,10 +154,10 @@ void WayPointsPendingResumptionHandler::on_event(
   }
 
   const smart_objects::SmartObject& response = event.smart_object();
-  const uint32_t corr_id = event.smart_object_correlation_id();
 
   SDL_LOG_TRACE("Received event with function id: "
-                << event.id() << " and correlation id: " << corr_id);
+                << event.id() << " and correlation id: "
+                << event.smart_object_correlation_id());
 
   auto current_pending = pending_requests_.front();
   pending_requests_.pop_front();

--- a/src/components/application_manager/src/resumption/resumption_data_processor_impl.cc
+++ b/src/components/application_manager/src/resumption/resumption_data_processor_impl.cc
@@ -1079,8 +1079,6 @@ void ResumptionDataProcessorImpl::CheckCreateWindowResponse(
     const smart_objects::SmartObject& request,
     const smart_objects::SmartObject& response) const {
   SDL_LOG_AUTO_TRACE();
-  const auto correlation_id =
-      response[strings::params][strings::correlation_id].asInt();
 
   const auto& msg_params = request[strings::msg_params];
   const auto app_id = msg_params[strings::app_id].asInt();
@@ -1093,8 +1091,9 @@ void ResumptionDataProcessorImpl::CheckCreateWindowResponse(
 
   const auto window_id = msg_params[strings::window_id].asInt();
   if (!IsResponseSuccessful(response)) {
-    SDL_LOG_ERROR("UI_CreateWindow for correlation id: " << correlation_id
-                                                         << " has failed");
+    SDL_LOG_ERROR("UI_CreateWindow for correlation id: "
+                  << response[strings::params][strings::correlation_id].asInt()
+                  << " has failed");
     auto& builder = application->display_capabilities_builder();
     builder.ResetDisplayCapabilities();
     return;

--- a/src/components/utils/src/convert_utils.cc
+++ b/src/components/utils/src/convert_utils.cc
@@ -34,6 +34,7 @@
 #include <stdint.h>
 #include <algorithm>
 #include <limits>
+#include <locale>
 #include "utils/macro.h"
 
 long long int utils::ConvertInt64ToLongLongInt(const int64_t value) {

